### PR TITLE
AddChain and fix LSpec infrastructure

### PR DESCRIPTION
**Pull request by mpenciak**
Adds some typeclasses and methods to calculate fast exponentiation.

### CODE REVIEW

**Comment by mpenciak** [comment]
I changed all the tests from inline `#lspec` to `lspecIO` so that they are part of the CI.

**Review by DanielRrr — APPROVED** [review]

File: YatimaStdLib/AddChain.lean [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+  mul ch₁ ch₂ := 
+    let last := ch₁.back
+    let ch₂' := ch₂.last.map (fun x => x * last) 
+    ch₁.append ch₂'
+
+/- 
+In this section we implement an efficient algorithm to calculate the minimal AddChain for a natural 
+number
+-/
+section nat
```
Comment (arthurpaulino) [inline YatimaStdLib/AddChain.lean:77]: Why start a section here?

**Review by arthurpaulino — APPROVED** [review]
The PR looks good, but we can just delete a lot of comments in the test files

File: YatimaStdLib/AddChain.lean (hunk 1 above)
Comment (mpenciak) [reply YatimaStdLib/AddChain.lean:77]: You're right, the extra section is superfluous with the namespace

**Pull request merged**